pkg/testworkflows/testworkflowexecutor/scheduler.go

```diff
@@ -260,24 +273,104 @@ func (s *scheduler) Schedule(ctx context.Context, sensitiveDataHandler Sensitive
 			return ch, err
 		}
 		for _, w := range list {
-			targets := execution.Targets
-
-			if isEmptyTargets(targets) && w.Spec.Execution != nil && w.Spec.Execution.Target != nil {
-				target := commonmapper.MapTargetApiToGrpc(w.Spec.Execution.Target)
-				targets = []*cloud.ExecutionTarget{target}
-			}
-
 			intermediateSelectors = append(intermediateSelectors, &cloud.ScheduleExecution{
 				Selector:      &cloud.ScheduleResourceSelector{Name: w.Name},
-				Targets:       targets,
+				Targets:       execution.Targets,
 				Config:        execution.Config,
 				ExecutionName: execution.ExecutionName, // TODO: what to do when execution name is configured, but multiple requested?
 				Tags:          execution.Tags,
 			})
 		}
 	}
 
+	// Determine and normalise the target for the scheduled executions
+	//
+	// Priority from high to low:
+	// - The target found directly on the scheduled request.
+	// - The target found on the Workflow Custom Resource.
+	// - The target found on any of the workflow's used templates. Throws an error if multiple define targets.
+	for i, _ := range intermediateSelectors {
+		exec := intermediateSelectors[i]
+		targets := exec.Targets
+
+		// Check the request for targets
+		if !isEmptyTargets(targets) {
+			continue
+		}
+
+		// Check the workflow for targets
+		workflows, err := testWorkflows.Get(exec.Selector)
+		if err != nil {
+			close(ch)
+			return ch, err
+		}
+
+		w, ok := lo.First(workflows)
+		if !ok {
+			close(ch)
+			return ch, err
+		}
```

Comment:
I think using `lo` here has caused an issue here in that it is both challenging to follow the logic flow, and you do not have an appropriate `err` to emit here (should be `nil` if all other error checks are working as expected).
```suggestion
		if len(workflows) == 0 {
			close(ch)
			return ch, errors.New("no workflows for target") // Maybe this should just be a straight `continue`?
		}
		// Should be safe to access because of the length check above.
		w := workflows[0]
```